server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingService.java

```diff
@@ -85,4 +105,23 @@ private boolean checkIfSameWeightsInMetadata(
     ) {
         return newWeightedRoutingMetadata.getWeightedRouting().equals(oldWeightedRoutingMetadata.getWeightedRouting());
     }
+
+    List<String> getAwarenessAttributes() {
+        return awarenessAttributes;
+    }
+
+    private void setAwarenessAttributes(List<String> awarenessAttributes) {
+        this.awarenessAttributes = awarenessAttributes;
+    }
+
+    public void verifyAwarenessAttribute(String attributeName) {
```

Comment:
Please add unit tests for this new method

Reply:
addressed